domain/unitstate/service/service.go

```diff
@@ -102,3 +121,12 @@ func (s *Service) SetState(ctx context.Context, as unitstate.AgentState) error {
 		return nil
 	})
 }
+
+// GetState returns the full agent state of the unit. The state may be empty.
+func (s *Service) GetState(ctx context.Context, uuid string) (unitstate.RetrievedUnitAgentState, error) {
+	state, err := s.st.GetUnitState(ctx, uuid)
```

Comment:
Why the rename? This seems like an unnecessary complication to me

Reply:
This is to match the `SetState` above.